### DISCUSSION TITLE
Hysteria2 在之後會支援meek嗎?

### DISCUSSION THREAD

**Discussion by TommcyOWO** [apernet/hysteria#1026] [Q&A]
自己有時還是會用到meek
不知道有沒有支援或以後可能?

自己目前在透過rust編寫基於quic和meek的協議，但想到已經有Hysteria了，不知道能不能綁在一起。

**Comment by haruue** [comment]
能详细描述一下这个需求吗？
没用过 meek， 看起来应该是 https://www.v2fly.org/v5/config/stream/meek.html
如果单纯是把 QUIC 协议产生 UDP 包封装进 TCP HTTP 里发送的话， 这种需求应该是不会得到支持。

**Reply by TommcyOWO** [reply]
meek原理就是meek client發出請求後，將加密後的訊息封裝成一個正常http request
meek server收到請求後解封裝，依照指示請求資料，但無法直接回傳給client
而client需要一直跟server要資料，有資料就回傳資料，沒有就回傳空白
每次最多只能反回64k的資料，且只能按順序接收
這讓meek一定快不了
我就想是否能夠用Hysteria 加速這過程，改變現狀
~~這只是我的想法，不知道可不可能~~

**Reply by haruue** [reply]
那看来 meek 的存在是为了能在一些较为严格的协议白名单的环境下使用。 我知道一些公司用防火墙限制员工上网， 只允许员工使用 HTTP 和 HTTPS， 或许还可能不允许长连接（例如限制每个 TCP 连接只能有一对请求和响应）。

这种环境下通常无法使用 UDP 相关的协议， 也因此不能使用 Hysteria。

Hysteria 通过定制的拥塞控制来实现加速， 强行把 Hysteria 的 UDP 包封装进真正的 TCP 连接里进行传输会导致 [TCP over TCP 问题](https://archive.is/4eKHE)， 会适得其反。

要在真正的 TCP 协议上使用 Hysteria 同款拥塞控制， 你可以试试在代理服务器上部署 [tcp-brutal](https://github.com/apernet/tcp-brutal)， 并参考它的 [开发者指南](https://github.com/apernet/tcp-brutal/blob/master/README.zh.md#%E5%BC%80%E5%8F%91%E8%80%85%E6%8C%87%E5%8D%97) 编写一个支持 tcp-brutal 的 meek 服务端以设置发送速率。 然而鉴于 meek 的设计以及它的运作环境， 我无法确保这能有很好的效果。

**Reply by TommcyOWO** [reply]
感謝回答
